synapse/handlers/presence.py

```diff
@@ -127,10 +130,10 @@ def __init__(self, hs: "HomeServer"):
         self.state = hs.get_state_handler()
 
         self._federation = None
-        if hs.should_send_federation() or not hs.config.worker_app:
+        if hs.should_send_federation():
             self._federation = hs.get_federation_sender()
```

Comment:
When we remove the federation send queue I plan on making `hs.get_federation_sender()` return an optional federation sender, which will simplify this even more.